index.js

```diff
@@ -1,122 +1,110 @@
-    shell.exit(1);
-  }
+  executeCommand("sudo apt-get install -y linux-headers-`uname -r`", "fail to install linux headers");
+}
+
+function installBcc(bcc_version) {
+  core.info(`Get BCC with version:` + bcc_version);
+  executeCommand("wget https://github.com/sustainable-computing-io/kepler-ci-artifacts/releases/download/v"+bcc_version+"/bcc_v"+bcc_version+".tar.gz", "fail to get BCC deb");
+  executeCommand("tar -zxvf bcc_v"+bcc_version+".tar.gz", "fail to untar BCC deb");
+  executeCommand("sudo dpkg -i libbcc_"+bcc_version+"-1_amd64.deb", "fail to install BCC deb");
+}
+
+function installXgboost(artifacts_version, xgboost_version) {
+  core.info(`Get xgboost with version:` + xgboost_version);
+  executeCommand("wget https://github.com/sustainable-computing-io/kepler-ci-artifacts/releases/download/v"+artifacts_version+"/xgboost-"+xgboost_version+"-Linux.sh.tar.gz", "fail to get xgboost pkg");
+  executeCommand("tar -zxvf xgboost-"+xgboost_version+"-Linux.sh.tar.gz", "fail to untar xgboost pkg");
+  executeCommand("sudo sh xgboost-"+xgboost_version+"-Linux.sh --skip-license  --prefix=/usr/local", "fail to install xgboost pkg");
+  executeCommand("sudo ldconfig", "fail to ldconfig");
+}
+
+function installLibbpf() {
   core.info(`libbpf`);
-  if (shell.exec("sudo apt-get install -y libbpf-dev").code !== 0) {
-    shell.echo("fail to install libbpf related package");
-    shell.exit(1);
-  }
+  executeCommand("sudo apt-get install -y libbpf-dev", "fail to install libbpf related package");
 }
 
-async function kubectl() {
-  let kubectl_version = core.getInput('kubectl_version');
-  core.debug(kubectl_version);
-  if (kubectl_version === undefined || kubectl_version == null || kubectl_version.length === 0) {
-    kubectl_version="1.25.4";
-  }
+function installKubectl(kubectl_version) {
   core.info(`Get kubectl with version `+ kubectl_version);
-  if (shell.exec("curl -LO https://dl.k8s.io/release/v"+kubectl_version+"/bin/linux/amd64/kubectl").code !== 0) {
-    shell.echo("fail to install kubectl");
-    shell.exit(1);
-  }
+  executeCommand("curl -LO https://dl.k8s.io/release/v"+kubectl_version+"/bin/linux/amd64/kubectl", "fail to install kubectl");
 }
 
 async function setup() {
-  let cluster_provider = core.getInput('cluster_provider');
-  core.debug(cluster_provider);
-  if (cluster_provider == undefined || cluster_provider == null || cluster_provider.length ===0){
-    cluster_provider="kind";
-  }
-  let local_dev_cluster_version = core.getInput('local_dev_cluster_version');
-  core.debug(local_dev_cluster_version);
-  if (local_dev_cluster_version === undefined || local_dev_cluster_version == null || local_dev_cluster_version.length === 0) {
-    local_dev_cluster_version="main";
-  }
+  const cluster_provider = getInputOrDefault('cluster_provider', 'kind');
+  const local_dev_cluster_version = getInputOrDefault('local_dev_cluster_version', 'main');
+  const prometheus_enable = getInputOrDefault('prometheus_enable', '');
```

Comment:
`''` or `false` ?

Reply:
'false' would be treated as a version string